src/ucp/core/ucp_request.inl

```diff
@@ -65,7 +66,7 @@ static UCS_F_ALWAYS_INLINE void
 ucp_request_put(ucp_request_t *req)
 {
     ucs_trace_req("put request %p", req);
-    ucs_assert(req->flags & UCP_REQUEST_FLAG_COMPLETED);
+    UCS_PROFILE_REQUEST_FREE(req);
```

Comment:
it's added because now ucp_request_put is called from other flows (to profile all places that request is released)